docs/code-examples/repository-pattern/01_simple_example/ilGeoLocationRepository.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+class ilGeoLocationRepository {
+
+    public const $table = 'geo_location';
+
+    public function __construct(ilDB $a_db)
+    {
+        $this->db = $a_db;
+    }
+
+    public function createGeoLocation(ilObjGeoLocation $obj)
+    {
+        $this->db->insert($this->table, array(
+            'id' => array('integer', $obj->getId()),

```

Comment:
I think this is a problem. How is a user of the repository supposed to get a valid id for her new object? I therefore prefer the pattern to pass the real data (`title`, ...) to `create` instead and determine a proper id for the new object internally in the repository.

This problem exists when we use consecutive ints, though, and could also be solved by using UUIDs.

Reply:
I agree with you. The dilemma I had at this point was following: Working with immutable objects is great. But what about objects without an id? Does this mean they don't exist yet? Should I recreate an immutable object just to set an id in the constructor? Since I don't use immutable objects very often, I had some struggle here.

Reply:
Let's discuss this in our call.